api/files/permissions.py

```diff
@@ -17,3 +19,13 @@ def has_object_permission(self, request, view, obj):
         return obj.checkout is None \
             or obj.checkout == auth.user \
             or obj.node.has_permission(auth.user, 'admin')
+
+
+class IsPreprintFile(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj):
+        assert isinstance(obj, FileNode), 'obj must be a FileNode, got {}'.format(obj)
+
+        if request.method not in permissions.SAFE_METHODS and len(Node.find(Q('preprint_file', 'eq', obj) & Q('_has_abandoned_preprint', 'eq', False))):
```

Comment:
This could be more efficient by checking the related keys from the object rather performing a query. Something like:

`[...] obj.node.preprint_file == obj and not obj.node._has_abandoned_preprint_file`